File: seunggil/kotlin/k진수에서 소수 개수 구하기.kt
class Solution {
    fun convert(n : Int, k: Int): String{
        var n = n
        val result = StringBuilder()
        while(n != 0){
            result.append(n % k)
            n = n / k
        }
        return result.reverse().toString()
    }

    fun isPrime(n : String) : Boolean{
        val n = n.toLong()
        if(n == 1L)
            return false
        for(i in 2 .. Math.sqrt(n.toDouble()).toInt()){
            if(n % i == 0L)
                return false
        }
        return true
    }

    fun solution(n: Int, k: Int): Int {

        val numbers = convert(n,k).split('0').filter {
            it.isNotEmpty()
        }

        var answer: Int = 0

        for(number in numbers){
            if(isPrime(number))
                answer++
        }
        return answer
    }
}

File: Jeonghak/Programmers/k진수에서소수개수구하기.cpp
#include <string>
#include <vector>
#include <iostream>
#include <cmath>

using namespace std;
	
bool isPrime(long n) {
    if(n==1) return false;
	for (int i = 2; i <= sqrt(n); i++) {
		if (n%i == 0) {
			return false;
		}
	}
	return true;
}

int solution(int n, int k) {
    int answer = 0;
    vector<long> converted;
    string temp = "";

    while(n>0){
        converted.push_back(n%k);
        n/=k;
    }
    
    for(int i=converted.size() - 1; i>=0; i--){
        if(converted[i] != 0){
            temp += converted[i] + '0';
        }
        if(converted[i] == 0 || i == 0){
            if(temp != "" && isPrime(stol(temp))){
                answer++;
            }
            temp = "";
        }
    }
    return answer;
}